### PR TITLE
Added new test cases

### PR DESCRIPTION
**Pull request by shubhammehta04**

### CODE REVIEW

File: cypress/e2e/pages/UserDashboard.js [hunk 1]
```diff
@@ -13,6 +13,10 @@ class UserDashboard extends BasePage {
   get editLastName() { return cy.get("input[value='User']") }
   get editMiddleName() { return cy.get("input[value='Mation']") }
   get updateButton() { return cy.get("button[class='btn btn-primary btn-sm']") }
+  get work() { return cy.get("h2[id='heading2'] button[type='button']") }
```
Comment (Sandy-N) [inline cypress/e2e/pages/UserDashboard.js:16]: add button and field postnames and short the css

File: cypress/e2e/pages/UserDashboard.js [hunk 2]
```diff
@@ -25,6 +29,26 @@ class UserDashboard extends BasePage {
     cy.log("Clicked on the Basic Info Option");
   }
 
+  clickOnWork() {
```
Comment (Sandy-N) [inline cypress/e2e/pages/UserDashboard.js:32]: on workdrd

File: cypress/e2e/pages/UserDashboard.js [hunk 3]
```diff
@@ -25,6 +29,26 @@ class UserDashboard extends BasePage {
...
+    this.personalDetails.click();
+    cy.log("Clicked on the Personal Details Option");
+  }
+
+  updateDOJ(doj) {
+    this.editDateOfJoining.type(doj).should('have.value', doj);
+    cy.log("Updated Date of Joining");
+  }
+
+  reverseDOJ(doj) {
```
Comment (Sandy-N) [inline cypress/e2e/pages/UserDashboard.js:47]: remove this

File: cypress/e2e/tests/EmployeeList.cy.js [hunk 4]
```diff
@@ -69,6 +78,34 @@ describe("Employee Onboard Tests", () => {
...
+        UserDashboard.getFieldValue("Date of Joining").should('equal', EmployeeData.DOJ);
+
+        //Verify that data do not get saved on clicking 'Update' button'
+        UserDashboard.clickOnEditButtonUnderWorkInfo();
+        UserDashboard.updateDOJ(EmployeeData.UpdateDOJ);
+        UserDashboard.clickOnUpdateButton();
+        UserDashboard.validateSuccessMessage();
+        UserDashboard.getFieldValue("Date of Joining").should('equal', EmployeeData.UpdatedDOJ);
+        UserDashboard.clickOnEditButtonUnderWorkInfo();
+        UserDashboard.reverseDOJ(EmployeeData.ReverseDOJ)
```
Comment (Sandy-N) [inline cypress/e2e/tests/EmployeeList.cy.js:103]: regular update method

File: cypress/e2e/pages/EmployeeListPage.js [hunk 5]
```diff
@@ -12,11 +13,18 @@ get department() { return cy.get("#department")}
...
   }  
 
+navigateToUserDashboardPage(nametxt){
+  sideBar.navigateTo("Employee Management", "Employees List");
+  this.searchByName.clear().type(nametxt).should('have.value', nametxt);
+  this.user.click();
+
+  }  
+
 validateNoRecordsAppear(){
```
Comment (parveen9719) [inline cypress/e2e/pages/EmployeeListPage.js:28]: Assertion must be in test class

File: cypress/e2e/pages/UserDashboard.js [hunk 6]
```diff
@@ -4,65 +4,104 @@ import Loaders from "../components/Loaders";
...
+  get editPermanentAddressTxt() { return cy.get("textarea[name='permanentAddress']:not(:disabled)") }
+  get editAlternateNumberTxt() { return cy.get("input[name='alternateNumber']:not(:disabled)") }
+  get editBloodGroupDrp() { return cy.get("div [id='bloodGroup']:not(:disabled)") }
+  get editMaritalStatusDrp() { return cy.get("select[id='maritalStatus']:not(:disabled)") }
+  get editGenderBtn() { return cy.get("#female:not(:disabled)") }
 
   // Methods
-  clickOnWorkExperience() {
-    this.workExperience.click();
+  clickOnWorkExperienceAccord() {
```
Comment (parveen9719) [inline cypress/e2e/pages/UserDashboard.js:32]: No need to define postfix like Accord in method name

Comment (shubhammehta04) [reply cypress/e2e/pages/UserDashboard.js:32]: Resolved

**Pull request merged**